bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/gson/IntesisHomeJSonDTO.java

```diff
@@ -0,0 +1,123 @@
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+
+/**
+ * {@link IntesisHomeJSonDTO} is used for the JSon/GSon mapping
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+public class IntesisHomeJSonDTO {
+
+    public static class Response {
+        public boolean success;
+        public JsonElement data;
+    }
+
+    public static class Data {
+        public JsonElement id;
+        public JsonElement info;
+        public JsonElement userinfo;
+        public JsonElement config;
+        public JsonElement dp;
+        public JsonElement dpval;
+    }
+
+    public static class Id {
+        public String sessionID; // Session ID
+    }
+
+    // Device Information used for thing properties
+    public static class Info {
+        public String wlanSTAMAC; // Device Client MAC Address
+        public String wlanAPMAC; // Device Access Point MAC Address
+        public String fwVersion; // Device Firmware Version
+        public String wlanFwVersion; // Wireless Firmware Version
+        public String acStatus; // Air Conditioner Communication Status
+        public String wlanLNK; // Connection Status with Wireless Network
+        public String ssid; // Wireless Network SSID
+        public String rssi; // Wireless Signal Strength
+        public String tcpServerLNK; // Cloud Server Connection (Not used for communication here)
+        public String localdatetime; // Local Date Time
+        public String powerStatus;
+        public String lastconfigdatetime; // Last Configuration Date Time
+        public String deviceModel; // Device Model
+        public String sn; // Serial Number
+        public String lastError;
+    }
+
+    public static class Userinfo {
+        public String username;
+        public JsonElement servicelist;
+    }
+
+    // List of available services
+    public static class Servicelist {
+        public String setconfig;
+        public String getconfig;
+        public String getcurrentconfig;
+        public String getinfo;
+        public String login;
+        public String logout;
+        public String passchange;
+        public String getavailabledatapoints;
+        public String setdatapointvalue;
+        public String getdatapointvalue;
+        public String getavailableservices;
+        public String reboot;
+        public String setdefaults;
+        public String getdefaultconfig;
+    }
+
+    public static class Config {
+        public String deviceModel; // Device Model
+        public String ip; // Device IP Address
+        public String netmask; // Device IP Address
+        public String dfltgw; // Default gateway
+        public boolean dhcp; // DHCP enabled
+        public String ssid; // WLAN Access Point
+        public int security; // Security Type
+        public int regdomain;
+        public int lastconfigdatetime;
+    }
+
+    public static class Dp {
+        public JsonArray datapoints; // dataPoints
```

Comment:
Any reason that you aren't using classes for the nested json structure?
```suggestion
        public Datapoints[] datapoints; // dataPoints
```

Reply:
I tried using classes, but the returned json is a bit malformed, so this was the simplest way to solve it.